### PR TITLE
[REVIEW]  Support GpuFilter and GpuCoalesceBatches for decimal data

### PR DESCRIPTION
**Pull request by sperlingxx**
Since both [PR](https://github.com/rapidsai/cudf/pull/6805) to enable `cudf::apply_boolean_mask` for decimal and [PR](https://github.com/rapidsai/cudf/pull/6615) to enable `cudf::contiguous_split` for decimal are ready, we can support `GpuCoalesceBatches` and `GpuFilter` (which relied on `GpuCoalesceBatches`).

This PR is to enable `GpuFilter` for decimal type. And related tests for `GpuFilter` and `GpuCoalesceBatches` are also provided.

In this PR, `isNotNull` is used as predicate of decimal filter because relational operators for decimal type are still unsupported.

### CODE REVIEW

**Comment by sperlingxx** [comment]
build

**Comment by revans2** [comment]
The integration tests started to crash after this change.  I am not 100% sure what happened but a lot of the tests started to get errors not being able to connect to the JVM process which makes me think that it crashed.

**Comment by revans2** [comment]
You can probably also look at `integration_tests/target/run_dir/target/surefire-reports/scala-test-detailed-output.log` if you do run in parallel mode, but sadly that is not available from the CI runs.

**Comment by sperlingxx** [comment]
@revans2 I also found integration tests failure of same pattern in PR #1191.

**Comment by revans2** [comment]
OK I am investigating it

**Comment by jlowe** [comment]
build

**Comment by sperlingxx** [comment]
build

**Comment by sperlingxx** [comment]
build

File: tests/src/test/scala/com/nvidia/spark/rapids/FilterExprSuite.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
```
Comment (jlowe) [inline tests/src/test/scala/com/nvidia/spark/rapids/FilterExprSuite.scala:2]:
```suggestion
 * Copyright (c) 2020, NVIDIA CORPORATION.
```

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Comment by sperlingxx** [comment]
build

**Comment by sperlingxx** [comment]
build

**Comment by sperlingxx** [comment]
build

**Comment by sperlingxx** [comment]
rebuild

**Comment by pxLi** [comment]
build

**Comment by sperlingxx** [comment]
build

**Comment by GaryShen2008** [comment]
build

**Review by revans2 — APPROVED** [review]

**Pull request merged**